pkg/pipeline/transform/transform_filter.go

```diff
@@ -159,25 +175,91 @@ func isRemoveEntrySatisfied(entry config.GenericMap, rules []*api.RemoveEntryRul
 	return true
 }
 
+func isKeepEntrySatisfied(entry config.GenericMap, rules []*api.KeepEntryRule) bool {
+	for _, r := range rules {
+		val, ok := entry[r.KeepEntry.Input]
+		switch r.Type {
+		case api.KeepEntryIfExists:
+			if !ok {
+				return false
+			}
+		case api.KeepEntryIfDoesntExist:
+			if ok {
+				return false
+			}
+		case api.KeepEntryIfEqual:
+			if !ok || val != r.KeepEntry.Value {
+				return false
+			}
+		case api.KeepEntryIfNotEqual:
+			if ok && val == r.KeepEntry.Value {
+				return false
+			}
+		case api.KeepEntryIfRegexMatch:
+			if ok {
+				match, ok := checkRegex(r.KeepEntry.Value, val)
+				if !ok || !match {
+					return false
+				}
+			} else {
+				return false
+			}
+		case api.KeepEntryIfNotRegexMatch:
+			if ok {
+				match, ok := checkRegex(r.KeepEntry.Value, val)
+				if !ok || match {
+					return false
+				}
+			} else {
+				return false
+			}
+		}
```

Comment:
Could we reuse the code from encode_prom_metric.go here ?

+ the implementations are differents since the prom one just negate the result of Regex function for the not match case which means the NotRegex function will return true if the field is missing.

Reply:
one other difference is that in prom filters api everything is string whereas here it can be anything, such as ints
using the native type of the field in the genericmap is more performant .. so maybe it's the prometheus api that I should align with this one for that matter..
I'll think more about that tomorrow!

Reply:
@jpinsonneau I finally did it: https://github.com/netobserv/flowlogs-pipeline/pull/740/commits/76131e0d8b973c4d3f0968e513536c13ef13f297